framework/src/parameters.rs

```diff
@@ -0,0 +1,172 @@
+            .parameters
+            .iter()
+            .map(|(name, _)| name.len())
+            .max()
+            .unwrap_or(0);
+
+        let name_delta = Point::new(6, 0);
+        let value_delta = name_delta + Point::new((max_name_width as i32 + 1) * 6, 0);
+
+        let style = MonoTextStyle::new(Font6x8, color);
+        let style_inverted = MonoTextStyleBuilder::new(Font6x8)
+            .background_color(color)
+            .build();
+
+        let mut position = Point::new(2, 2);
+
+        for (index, (name, value)) in self.parameters.iter().enumerate() {
+            if index == self.selected {
+                Text::new(">", position).into_styled(style).draw(target)?;
+            }
+
+            if index == self.selected && self.active {
+                Text::new(name, position + name_delta)
+                    .into_styled(style_inverted)
+                    .draw(target)?;
+            } else {
+                Text::new(name, position + name_delta)
+                    .into_styled(style)
+                    .draw(target)?;
+            }
+
+            Text::new(&value.to_string(), position + value_delta)
+                .into_styled(style)
+                .draw(target)?;
+
+            position.y += 8;
+        }
+
+        Ok(())
+    }
+
+    pub(crate) fn handle_events(&mut self, window: &mut Window) -> bool {
+        for event in window.events() {
+            let event = match event {
+                SimulatorEvent::Quit => return true,
+                SimulatorEvent::KeyDown { keycode, .. } => match keycode {
+                    Keycode::Up => Event::Up,
+                    Keycode::Down => Event::Down,
+                    Keycode::Left => Event::Left,
+                    Keycode::Right => Event::Right,
+                    Keycode::Space | Keycode::Return => Event::Activate,
+                    _ => continue,
+                },
+                SimulatorEvent::MouseButtonDown { mouse_btn, point }
+                    if mouse_btn == MouseButton::Left =>
+                {
+                    self.mouse_button_down = true;
+                    Event::MouseMove(point)
+                }
+                SimulatorEvent::MouseButtonDown { mouse_btn, .. }
+                    if mouse_btn == MouseButton::Middle =>
+                {
+                    Event::Activate
+                }
+                SimulatorEvent::MouseMove { point } if self.mouse_button_down => {
+                    Event::MouseMove(point)
+                }
+                SimulatorEvent::MouseButtonUp { .. } => {
+                    self.mouse_button_down = false;
+                    continue;
+                }
+                SimulatorEvent::MouseWheel {
+                    mut scroll_delta,
+                    direction,
+                } => {
+                    if direction == MouseWheelDirection::Flipped {
+                        scroll_delta = scroll_delta.component_mul(Point::new(-1, -1));
```

Comment:
Could you try to remove this line to see if it fixes the scroll direction? Or, if it doesn't, tell me the value for direction on your system?

Reply:
Where did you notice that the scroll direction was wrong? Only when you tried to change an integer value or also when you navigated the menu? The problem might have been fixed by the new version, if it only occurred for integer values.

Reply:
With a local checkout of 31b2370, if I scroll "down" (i.e. pull the top of the scroll wheel towards me), the menu items cycle backwards (bottom -> top). If I activate a value and scroll in the same direction, the value increases. I'm ambivalent about the value direction, but the menu item scroll direction remains backwards.

Interestingly, if I comment out the `scroll_delta = scroll_delta.component_mul(Point::new(-1, -1));` line, the menu scroll direction doesn't change. The arrow keys work as I'd expect.

Scrolling is a nice interaction, but if it's inconsistent between hardware, the simpler solution may be to leave it out and just use keyboard interaction for now.

Reply:
I also noticed that running the `line` debugger, pressing down on the arrow key increases the stroke width which is the opposite of what I'd expect.

Reply:
> I also noticed that running the `line` debugger, pressing down on the arrow key increases the stroke width which is the opposite of what I'd expect.

This should have been fixed in 31b2370.

Reply:
Ah my bad, I don't think I was on the latest changes. That is indeed fixed.

Reply:
After looking at the SDL source code, I found out that the direction parameter is only supported on Mac OS. I've removed the mouse wheel support because there doesn't seem to be a way to get a consistent behavior across different configurations.